Dockerfile

```diff
@@ -14,10 +14,4 @@ RUN apt-get update && \
 RUN cargo build --release --target=x86_64-unknown-linux-musl && \
     cp target/x86_64-unknown-linux-musl/release/rust-representer ./bin/
 
-FROM alpine:latest
```

Comment:
Deleting this is a problem. This line creates a new... layer? stage? I'm not sure exactly what it's called. But essentially, everthing after it creates an image separate from what came before it.

The reason we do this here is we don't want to keep everthing around from the build in terms of storage space. All the tooling containers combined need some amount of storage space, which every server needs to have at least. So keeping the size of these images small contributes to lower server costs for exercism.

Judging from your commit message, the reason you did this is because the container didn't find `rustfmt`. In that case, we need to find a way to install rustfmt in the alpine image, without wasting space.

Reply:
I don't have time to investigate the best way to do that. The releases page of `rustfmt` doesn't have prebuilt binaries for our target.

Two options I can think of right now:

Clone the [rustfmt repo](https://github.com/rust-lang/rustfmt) in the build stage, compile from source for our target, copy the binary over to the alpine stage

Maybe `rustup` can be coerced into downloading tooling for a different target than the one of the current machine? Then just copy it over. That would probably be the fastest way. But I just investigated this, it looks like rustup installes its components as dynamically linked executables, I don't think we want to mess with that.

Reply:
The thing is this, installing the entire Rust toolchain just to have `rustfmt` is way too much. One toolchain is about 500 MiB, and I personally don't know the best way to strip that down to only what we need.

Reply:
Building from source might be the best option. To save ourselves the trouble of cross-compiling, we could change the build stage base from debian to alpine as well :thinking:

Reply:
Ok I will try to readd this step today. Thanks for taking time to search on it !

Reply:
What about running rustfmt on the rust binary directly using it as a dependency to avoid having to install it like this code : https://github.com/bcmyers/num-format/blob/v0.1.2/num-format-dev/src/rustfmt.rs

Reply:
Yes, that would be a very clean solution. It also entails compiling `rustfmt` from source, so we still incur the additional build time. But both `run.sh` and the dockerfile can be simplified greatly with this approach.

Not all programs that are primarily meant to be run as standalone executables have good support for being embedded as a library. I'm optimistic, but it must be investigated.

Reply:
Ok I will try today ! Thanks for taking time to answer me on this PR :)

Reply:
As you expected it's quite hard to run rustfmt as a library (in stable) but I found this project made by the syn maintainer (used by syn internally) : https://github.com/dtolnay/prettyplease . It seems to be a perfect fit for us as we are managing syn AST.

What do you think about removing the first and second call to rustfmt to only one call in the rust code to this library only AFTER the code has been normalized ?

Reply:
Yes that sounds great!

Reply:
It works well now I have the same dockerfile as before and added the [`run-in-docker.sh`](https://github.com/exercism/python-representer/blob/main/bin/run-in-docker.sh) from `python-representer` to make local testing using docker more easy.